src/datasets/arrow_dataset.py

```diff
@@ -1655,10 +1655,7 @@ def __iter__(self):
         selected format.
         """
         for index in range(self.num_rows):
-            yield self._getitem(
-                index,
-                decoded=False,
-            )
+            yield self._getitem(index)
```

Comment:
As discussed, I remember I set `decoded=False` here to avoid decoding just by iterating over examples of dataset. We wanted to decode only if the "audio" field (for Audio feature) was accessed.

Reply:
That is why the CI Test Audio fails after this modification: https://github.com/huggingface/datasets/runs/4608657230?check_suite_focus=true
```
=========================== short test summary info ============================
FAILED tests/features/test_audio.py::test_dataset_with_audio_feature_map_is_not_decoded
FAILED tests/features/test_audio.py::test_dataset_with_audio_feature_map_is_decoded
========================= 2 failed, 15 passed in 8.37s =========================